tests/flows/test_full_pipeline.py

```diff
@@ -122,20 +121,33 @@ def test_validate_aggregation_inference_configs_raises_value_error(
     assert expected_error in str(e.value)
 
 
+class MockFlowWithOptions:
+    """Mock flow with options that returns a given value."""
+
+    def __init__(self, return_value):
+        """Mock __init__ method that sets the return value."""
+        self.return_value = return_value
+
+    async def __call__(self, **kwargs):
+        """Mock __call__ method that returns the return value."""
+        return self.return_value
```

Comment:
Could add functionality to store call args but this seems to be re-inventing the wheel.